cumulus/client/cli/src/lib.rs

```diff
@@ -340,6 +352,14 @@ impl RunCmd {
 
 		CollatorOptions { relay_chain_mode }
 	}
+
+	/// Create [`DhtBootnodeOptions`] enabling embedded DHT bootnode & DHT bootnode discovery.
+	pub fn dht_bootnode_options(&self) -> DhtBootnodeOptions {
```

Comment:
You can just put this into the collator options. The naming was not so nice from my side, its not only for collators :grimacing:.

Reply:
Done in [`3ae5b94`](https://github.com/paritytech/polkadot-sdk/pull/8072/commits/3ae5b94d52b6436d62f1a6d612b3b0acfe537e42).